Homework/homework1/task5.py

```diff
@@ -0,0 +1,20 @@
+"""
+Given a list of integers numbers "nums".
+
+You need to find a sub-array with length less equal to "k", with maximal sum.
+The written function should return the sum of this sub-array.
+Examples:
+    nums = [1, 3, -1, -3, 5, 3, 6, 7], k = 3
+    result = 16
+"""
+from typing import List
+
+
+def find_maximal_subarray_sum(nums: List[int], k: int) -> int:
+    if k > len(nums) or not nums or k == 0:
```

Comment:
`k > len(nums)`  which part of the task description is related to this check?

Reply:
Thats a point there is nothing about it. Or we assume that the function will be used properly and 'k' will be less than length of nums list?

Reply:
Why do you have tests with different `k` and the same result?

```
 ([-1, 2, -4, 5, 6, -9, 0], 3, 11),
 ([-1, 2, -4, 5, 6, -9, 0], 4, 11),
 ([-1, 2, -4, 5, 6, -9, 0], 5, 11),
```

Reply:
Because we need to find a sub-array <= k with the max sum.
For k=3 - (k=1: max_sum=6; k=2: max_sum=11; k=3: max_sum=7 {-4, 5, 6})
For k = 4 - (the same; k=4: max_sum=9 {2, -4, 5, 6}) and etc.
So just to be sure that the function really find the sub-array <= k with max sum, not with sub array = k with max sum.
Or its too much and too suspicious again that it doesnt change?

Reply:
Lest back to the initial question, can we get an answear with `k > len(nums)`?

Reply:
Oh ops, I got it. So yes, we can have an answer but 'k' value wouldnt reach and check of sums will stop on k = length of nums.